src/sentry/api/endpoints/project_dynamic_sampling.py

```diff
@@ -0,0 +1,194 @@
+        :auth: required
+        """
+        if not features.has(
+            "organizations:filters-and-sampling", project.organization, actor=request.user
+        ):
+            return Response(
+                {
+                    "detail": [
+                        "Dynamic sampling feature flag needs to be enabled before you can perform "
+                        "this action."
+                    ]
+                },
+                status=404,
+            )
+
+        query = request.GET.get("query", "")
+        requested_sample_size = min(int(request.GET.get("sampleSize", 100)), 1000)
+        distributed_trace = request.GET.get("distributedTrace", "1") == "1"
+        stats_period = min(
+            parse_stats_period(request.GET.get("statsPeriod", "1h")), timedelta(hours=24)
+        )
+
+        end_time = timezone.now()
+        start_time = end_time - stats_period
+
+        # Fetches a random sample of root transactions of size `sample_size` in the last period
+        # defined by `stats_period`. The random sample is fetched by ordering by the
+        # `random_number` which is generated by a `random_number` function which generates a random
+        # number for every row. The goal here is to fetch the transaction ids, their sample rates
+        # and their trace ids.
+        root_transactions = discover.query(
+            selected_columns=[
+                "id",
+                "trace",
+                "trace.client_sample_rate",
+                "random_number() AS random_number",
+            ],
+            query=f"{query} event.type:transaction !has:trace.parent_span_id",
+            params={
+                "start": start_time,
+                "end": end_time,
+                "project_id": [project.id],
+                "organization_id": project.organization,
+            },
+            orderby=["-random_number"],
+            offset=0,
+            limit=requested_sample_size,
+            equations=[],
+            auto_fields=True,
+            auto_aggregations=True,
+            allow_metric_aggregates=True,
+            use_aggregate_conditions=True,
+            transform_alias_to_input_format=True,
+            functions_acl=["random_number"],
+            referrer="dynamic-sampling.distribution.fetch-parent-transactions",
+        )["data"]
+
+        sample_size = len(root_transactions)
+        sample_rates = sorted(
+            transaction.get("trace.client_sample_rate") for transaction in root_transactions
+        )
+        if len(sample_rates) == 0:
+            return Response(
+                {
+                    "project_breakdown": None,
+                    "sample_size": sample_size,
+                    "null_sample_rate_percentage": None,
+                    "sample_rate_distributions": None,
+                }
+            )
+
+        non_null_sample_rates = sorted(
+            float(sample_rate) for sample_rate in sample_rates if sample_rate not in {"", None}
```

Comment:
I think you can totally do this computation in a clickhouse sql expression, including converting the string to a float. But I suppose you can't funnel it through discover